modules/tensor_mechanics/src/materials/ComputeElasticityTensorCP.C

```diff
@@ -0,0 +1,78 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ComputeElasticityTensorCP.h"
+#include "RotationTensor.h"
+
+template<>
+InputParameters validParams<ComputeElasticityTensorCP>()
+{
+  InputParameters params = validParams<ComputeElasticityTensorBase>();
+  params.addClassDescription("Compute an elasticity tensor.");
+  params.addRequiredParam<std::vector<Real> >("C_ijkl", "Stiffness tensor for material");
+  params.addParam<MooseEnum>("fill_method", RankFourTensor::fillMethodEnum() = "symmetric9", "The fill method");
+  params.addParam<UserObjectName>("read_prop_user_object","The ElementReadPropertyFile GeneralUserObject to read element specific property values from file");
+  return params;
+}
+
+ComputeElasticityTensorCP::ComputeElasticityTensorCP(const InputParameters & parameters) :
+    ComputeElasticityTensorBase(parameters),
+    _Cijkl(getParam<std::vector<Real> >("C_ijkl"), (RankFourTensor::FillMethod)(int)getParam<MooseEnum>("fill_method")),
```

Comment:
My vote is for having the rotation w.r.t. Euler angles in a separate ComputeElasticityTensor class as well! If this is done cleverly it will allow easy extension to variable dependent elasticity tensors. This could be powerful.